### ISSUE TITLE
Sortable Pages stop working now [Grav v1.3.0 / Admin v1.5.0]

### ISSUE THREAD

**Issue by ash0080** [getgrav/grav#1574]
Can not save the order

**Comment by ash0080** [comment]
Noticed that the folder numeric prefix is still working. means change the order in page editor will rename the folders, but Grav can't show them as before. both on frontend and backend

**Comment by timrobbings** [comment]
I am also seeing this behaviour after updating to Grav v1.3.0/Admin v1.5.0.

I _think_ I can see the problem, see [this PR for my proposed fix](https://github.com/getgrav/grav/pull/1575).

Although admittedly, I have a fairly rudimentary understanding of the codebase so people can let me know if I've missed something!

**Comment by MattAppleton** [comment]
Second this, page ordering was working with rc05, newest Admin seems to break it, have tried clearing cache and going again but not working.

**Comment by ash0080** [comment]
@timrobbings It works, Great!

**Comment by rhukster** [comment]
merged, will release later tonight or tomorrow

**Comment by ash0080** [comment]
@rhukster wait, seems still has problem, the fix doesn't work after touching the page in editor again.

**Comment by rhukster** [comment]
if you can provide a PR with a fix that would be appreciated. I don't have any access right now to a linux system that exhibits this behavior.

**Comment by timrobbings** [comment]
I am having some difficulty reproducing the issue but I am happy to have another look later if you can provide more info on the exact sequence of steps you are taking.

**Comment by ash0080** [comment]
![2017-07-18 4 17 19](https://user-images.githubusercontent.com/2803210/28307219-bd03e10c-6bd4-11e7-839d-9f08f274931c.png)

After dragging


![2017-07-18 4 17 27](https://user-images.githubusercontent.com/2803210/28307223-c0232334-6bd4-11e7-934e-231ba5460df7.png)

After saving

**Comment by zhedar** [comment]
Can confirm this, sadly the fix proposed in #1575 doesn't have any effect. I also tried upgrading to core 1.3.0 alone, thus without updating the admin plugin. Still had the same problem though. 

It seems like the pages are ordered by the date of creation. If I create a new page manually or through the admin plugin, it is still numbered correctly but not sorted after the other pages. This happens both in the admin view and the frontend.
<img width="239" alt="bildschirmfoto 2017-07-18 um 14 51 25" src="https://user-images.githubusercontent.com/6633403/28319556-7b777baa-6bce-11e7-99c7-b19f7bce474e.png">

**Comment by flaviocopes** [comment]
Uhm.. I tried replicating on 2 different setups (osx and linux) but I cannot recreate the issue.

**Comment by zhedar** [comment]
I'm on macOS as well, using the most recent version of MAMP (MAMP 4.11, Apache, PHP 7.11). Maybe it's also important, that I'm using a internationalized project as that seems to effect sorting as well.

**Comment by ash0080** [comment]
I made a clean reinstall using MAMP with all default configs, then the patch, seems not work at all,
I remember a moment that seemed to be successful,  but I can't reproducing the success now

**Comment by timrobbings** [comment]
I've just had another look at this. I'm using Ubuntu, PHP 7.0.18 and Apache.

So far I've tried inserting unusual characters (accented, hieroglyphic, etc...) and tried removing leading zeros from folder names, but I'm still not having much success.

@ash0080 / @zhedar  - what are the folder names (as displayed if you list the content of the directory)?

Could you paste the names so that I can copy them and have another go?

**Comment by matejkramny** [comment]
How does grav determine the order of the pages if not in alphabetical order?

**Comment by timrobbings** [comment]
@zhedar:

> If I create a new page manually or through the admin plugin, it is still numbered correctly but not sorted after the other pages. This happens both in the admin view and the frontend.

My understanding is that this is intentional behaviour - looking at the code, it appears manually ordered items are inserted _before_ those with other ordering criteria. I think this is right but perhaps someone could confirm?

**Comment by zhedar** [comment]
@timrobbings 
That's right. All my items have numeric sorting enabled and therefore are manually ordered. Therefore they get numeric prefixes, don't they? But as you can see in the screenshot above, this sorting doesn't apply correctly anymore. I'm currently using a backup version of 1.2.4 which still works as intended(?).

**Comment by timrobbings** [comment]
@zhedar - what do your folder _paths_ look like?

**Comment by zhedar** [comment]
@timrobbings Here's a full overview of all my folder names and the first level's contents (I've got 2 levels of modular pages):

```
file  user/pages/home/*/*
user/pages/home/01._showcase/header2.jpg:        JPEG image data
user/pages/home/01._showcase/showcase.de.md:     ASCII text
user/pages/home/02._features/features.de.md:     UTF-8 Unicode text
user/pages/home/03._service/01._header:          directory
user/pages/home/03._service/02._subtext:         directory
user/pages/home/03._service/03._mitarbeiter:     directory
user/pages/home/03._service/04._mitarbeiter2:    directory
user/pages/home/03._service/05._service-text:    directory
user/pages/home/03._service/modular.de.md:       UTF-8 Unicode text
user/pages/home/04._teiledienst/01._header:      directory
user/pages/home/04._teiledienst/02._mitarbeiter: directory
user/pages/home/04._teiledienst/modular.de.md:   ASCII text
user/pages/home/05._verkauf/01._header:          directory
user/pages/home/05._verkauf/02._subtext:         directory
user/pages/home/05._verkauf/03._verkaufsgebiet:  directory
user/pages/home/05._verkauf/04._mitarbeiter:     directory
user/pages/home/05._verkauf/05._mitarbeiter2:    directory
user/pages/home/05._verkauf/modular.de.md:       ASCII text
user/pages/home/06._vermietung/01._header:       directory
user/pages/home/06._vermietung/02._subtext:      directory
user/pages/home/06._vermietung/03._mitarbeiter:  directory
user/pages/home/06._vermietung/employees.de.md:  ASCII text
user/pages/home/09._karriere/14276-049.jpg:      JPEG image data
user/pages/home/09._karriere/text.de.md:         exported SGML document text, UTF-8 Unicode text
user/pages/home/10._kontakt/_anschrift:          directory
user/pages/home/10._kontakt/_kontaktdaten:       directory
user/pages/home/10._kontakt/_oeffnungszeiten:    directory
user/pages/home/10._kontakt/flexcolumns.de.md:   UTF-8 Unicode text
```

**Comment by zhedar** [comment]
Also just tried to deactivate all i18n features and remove some of the `.de` suffixes to make the site work in general without i18n, but that didn't change anything that far. Might have overlooked something important there though as it was only a quick test. Also didn't have the fix from #1575 active during that test run. May have another try soon.

**Comment by timrobbings** [comment]
Thanks @zhedar.

FWIW, I am seeing problems with subfolders - which may be related although I'm not sure if it is the same as the one you describe. I'll summarise here.

Specifically, when you have the following structure:
```
01.home
  01.featured-work
02.portfolio
  01.item-1
  02.item-2
  03.item-3
```
If you configure the `01.featured-work` template (a module of `01.home`) to render a subset of `02.portfolio`, then you should see that the collection _is not_ correctly ordered.

Whereas, if you view the `02.portfolio` template, you should see that the collection _is_ correctly ordered.

Perhaps suggesting that sort order is not conveyed or applied correctly when pulling collections from sibling directory trees.

**Comment by zhedar** [comment]
Sorry, I feel a bit foolish right now. As I wrote I incorporated the fix from #1575, but I just looked into that files history and saw that #1571 made another important change, which I didn't seem to have as I didn't add another git remote just for a single character change. Somehow this one wasn't referenced here yet, although I read about it in a comment at the blog post.

After adding these two changes sorting works as before 1.3 again. Will do some tests.

**Comment by timrobbings** [comment]
No problem, appreciate the help :+1: 

I think I've tracked down a potential issue relating to sorting as mentioned above, which probably isn't helping matters - I will also try and do some more testing and raise a separate ticket to avoid conflating multiple issues.

**Comment by ash0080** [comment]
![2017-07-19 3 20 20](https://user-images.githubusercontent.com/2803210/28355247-cb35afe6-6c95-11e7-9279-90140f833ea5.png)
Clean reinstalled test folders

**Comment by flaviocopes** [comment]
Can you confirm the latest `develop`  has solved the problem for you? I still cannot replicate it.

**Comment by MattAppleton** [comment]
Yes, confirmed fixed for me, I think I tried to confirm this via Slack too, perhaps not the right way, sorry!



thanks
Matt


- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
modern activity, Buzzer #3, 19 Warburton Road, London E8 3RT 
T: +44 (0)208 985 6756 / M: +44 (0)7941 806 766
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

**Comment by flaviocopes** [comment]
Fixed in 1.3.1 🚢
